src/jenkins.py

```diff
@@ -285,6 +296,35 @@ def _setup_user_token(self, container: ops.Container) -> None:
             container.push(API_TOKEN_PATH, token, user=USER, group=GROUP)
         except ops.pebble.PathError as exc:
             raise JenkinsBootstrapError("Failed to setup user token.") from exc
+        except jenkinsapi.utils.requester.JenkinsAPIException as e:
+            # Jenkins api exception when generating user token
+            # We check if security is disabled
+            logger.info("Generate token failed, checking if security is disabled")
+            response = client.requester.get_url(
+                f"{client.base_server_url()}/manage/api/json?tree=useSecurity"
+            )
+            try:
+                if response.status_code == 200 and not response.json()["useSecurity"]:
+                    # !! Write a random string to the api token as a temporary workaround,
```

Comment:
What are you working around with this? Why is this needed?

Reply:
yes, like we discussed before, the API token generation would need to be refactored so that the token is generated on-demand. But I thought it'll be too big of a change to implement it here so I added this as a temporary fix before doing a follow-up PR to refactor the token generation.